tools/tgentools/tgentools

```diff
@@ -283,11 +283,19 @@ def analyze(args):
         args.nprocesses = cpu_count()
 
     if args.nprocesses > 1:
-        analysis = ParallelAnalysis(nickname=args.nickname, ip_address=args.ip_address)
-        analysis.analyze(paths, do_complete=args.do_complete, date_filter=args.date_filter,
-            num_subprocs=min(args.nprocesses, len(paths)))
+        num_subprocs=min(args.nprocesses, len(paths))
+        analysis = ParallelAnalysis(
+            nickname=args.nickname,
+            ip_address=args.ip_address,
+            num_subprocs=num_subprocs,
+        )
+        analysis.analyze(paths, do_complete=args.do_complete, date_filter=args.date_filter)
     else:
-        analysis = SerialAnalysis(nickname=args.nickname, ip_address=args.ip_address)
+        analysis = SerialAnalysis(
+            nickname=args.nickname, 
+            ip_address=args.ip_address,
+            num_subprocs=1,
+        )
         analysis.analyze(paths, do_complete=args.do_complete, date_filter=args.date_filter)
```

Comment:
I think this also limits the xz parallelism to the number of files to analyze? So if you only provide one file, xz will only use 1 thread? Is this restriction needed?

Reply:
Good point! I changed the behavior so it does not limit the number of xz threads based on the number of files being processed.